can/interfaces/neousys_wdt.py

```diff
@@ -0,0 +1,269 @@
+    ]
+
+
+# valid:2~16, sum of the Synchronization, Propagation, and
+#             Phase Buffer 1 segments, measured in time quanta.
+# valid:1~8, the Phase Buffer 2 segment in time quanta.
+# valid:1~4, Resynchronization Jump Width in time quanta
+# valid:1~1023, CAN_CLK divider used to determine time quanta
+class NeousysWdtCanBitClk(Structure):
+    """ C CAN BIT Clock struct """
+
+    _fields_ = [
+        ("syncPropPhase1Seg", c_ushort),
+        ("phase2Seg", c_ushort),
+        ("jumpWidth", c_ushort),
+        ("quantumPrescaler", c_ushort),
+    ]
+
+
+WDT_CAN_MSG_CALLBACK = CFUNCTYPE(None, POINTER(NeousysWdtCanMsg), c_uint)
+WDT_CAN_STATUS_CALLBACK = CFUNCTYPE(None, c_uint)
+
+WDT_CAN_MSG_EXTENDED_ID = 0x0004
+WDT_CAN_MSG_REMOTE_FRAME = 0x0040
+WDT_CAN_MSG_DATA_NEW = 0x0080
+WDT_CAN_MSG_DATA_LOST = 0x0100
+
+WDT_CAN_MSG_USE_ID_FILTER = 0x00000008
+WDT_CAN_MSG_USE_DIR_FILTER = (
+    0x00000010 | WDT_CAN_MSG_USE_ID_FILTER
+)  # only accept the direction specified in the message type
+WDT_CAN_MSG_USE_EXT_FILTER = (
+    0x00000020 | WDT_CAN_MSG_USE_ID_FILTER
+)  # filters on only extended identifiers
+
+WDT_CAN_STATUS_BUS_OFF = 0x00000080
+WDT_CAN_STATUS_EWARN = (
+    0x00000040  # can controller error level has reached warning level.
+)
+WDT_CAN_STATUS_EPASS = (
+    0x00000020  # can controller error level has reached error passive level.
+)
+WDT_CAN_STATUS_LEC_STUFF = 0x00000001  # a bit stuffing error has occurred.
+WDT_CAN_STATUS_LEC_FORM = 0x00000002  # a formatting error has occurred.
+WDT_CAN_STATUS_LEC_ACK = 0x00000003  # an acknowledge error has occurred.
+WDT_CAN_STATUS_LEC_BIT1 = (
+    0x00000004  # the bus remained a bit level of 1 for longer than is allowed.
+)
+WDT_CAN_STATUS_LEC_BIT0 = (
+    0x00000005  # the bus remained a bit level of 0 for longer than is allowed.
+)
+WDT_CAN_STATUS_LEC_CRC = 0x00000006  # a crc error has occurred.
+WDT_CAN_STATUS_LEC_MASK = (
+    0x00000007  # this is the mask for the can last error code (lec).
+)
+
+
+class NeousysWdtBus(BusABC):
+    """ Neousys WDT_DIO Canbus Class"""
+
+    def __init__(self, channel, device=0, bitrate=500000, **kwargs):
+        """
+        :param channel: channel number
+        :param device: device number
+        :param bitrate: bit rate. Renamed to bitrate in next release.
+        """
+        super(NeousysWdtBus, self).__init__(channel, **kwargs)
+
+        self.canlib = None
+
+        try:
+            if platform.system() == "Windows":
+                self.canlib = WinDLL("./WDT_DIO.dll")
+            else:
+                self.canlib = CDLL("libwdt_dio.so")
+
+            logger.info("Loaded Neousys WDT_DIO Can driver")
+
+            self.channel = channel
+
+            self.device = device
+
+            self.channel_info = "Neousys WDT_DIO Can: device {}, channel {}".format(
+                self.device, self.channel
+            )
+
+            self.lock = threading.Lock()
+            self.recv_msg_array = []
```

Comment:
Is there any reason why you are using a Lock + List rather than a [Queue](https://docs.python.org/3/library/queue.html#module-queue)?